File: 151001/Shlyk/SpringCalculator/spring-calculator-rest/src/main/java/by/bsuir/poit/dc/rest/model/NewsLabel.java
package by.bsuir.poit.dc.rest.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author Paval Shlyk
 * @since 31/01/2024
 */
@Entity
@Table(name = "news_label")
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class NewsLabel {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "news_id", referencedColumnName = "id")
    private News news;

    @ManyToOne
    @JoinColumn(name = "label_id", referencedColumnName = "id")
    private Label label;
}

File: 151004/Bosko/lab2/lab2/Models/NewsLabel.cs
namespace lab2.Models
{
    public partial class NewsLabel
    {
        public int Id { get; set; }

        public int? NewsId { get; set; }

        public int? LabelId { get; set; }

        public News? News { get; set; }

        public Label Label { get; set; }
    }
}
